pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -208,7 +208,7 @@ func (c *Cluster) Start(clusterInfo *cephclient.ClusterInfo, rookVersion string,
 	logger.Infof("start running mons")
 
 	logger.Debugf("establishing ceph cluster info")
-	if err := c.initClusterInfo(cephVersion); err != nil {
+	if err := c.initClusterInfo(cephVersion, c.ClusterInfo.NamespacedName().Name); err != nil {
```

Comment:
I don't think this is going to work. You cannot initialize clusterInfo.Name by passing `c.ClusterInfo.NamespacedName().Name`, since the name has not been set yet. Or am I missing something?

Reply:
They are initialized first at https://github.com/rook/rook/blob/c3b5d3b078f096e6eaa5d0b5d0fcbe1206b0b071/pkg/operator/ceph/cluster/mon/mon.go#L190 , then re-initialized.

Elaborated it[ here in the issue.](https://github.com/rook/rook/issues/9159#issue-1052184109).

And checked it, works for the future method calls.